micrometer-observation/src/main/java/io/micrometer/observation/Observation.java

```diff
@@ -380,8 +428,8 @@ default boolean isNoop() {
 
     /**
      * Adds an observation convention that can be used to attach key values to the
```

Comment:
nit: change `Adds` to `Sets`.

Also, do we want to deprecate this method as we discussed offline?

The actual implementation only set the convention if it supports the context. Wondering should it also be documented?

Reply:
Fixed in https://github.com/micrometer-metrics/micrometer/commit/3f0ca2f44b6b1fd8a343c1efd9292f63eb0d6f07, thanks for catching it.

I think we should deprecate it, but:
- I think it should happen in 1.11
- We need more discussions about the potential use-cases and consequences

I added a note about setting the convention in https://github.com/micrometer-metrics/micrometer/commit/22a81f0cf1ac3f21e3b0b1f93a2b8c5c079422e2 thank you for noticing it.